frontend/src/pages/pipelines/global/modelCustomization/baseModelSection/InlineEditText.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import {
+  TextInput,
+  Button,
+  Flex,
+  FlexItem,
+  Content,
+  Alert,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { PencilAltIcon, CheckIcon, TimesIcon } from '@patternfly/react-icons';
+
+interface InlineEditTextProps {
+  text: string;
+  onSave: (text: string) => void;
+  checkSupported: (text: string) => boolean;
+  unsupportedMessage: string;
+}
+
+const InlineEditText: React.FC<InlineEditTextProps> = ({
+  text,
+  onSave,
+  checkSupported,
+  unsupportedMessage,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(text);
+  const [showUnsupportedMessage, setShowUnsupportedMessage] = useState(false);
+
+  const isEmpty = React.useMemo(() => inputValue.trim().length === 0, [inputValue]);
+
+  React.useEffect(() => {
+    setShowUnsupportedMessage(!(isEditing || isEmpty || checkSupported(inputValue)));
```

Comment:
Checking `isEmpty` is not required, because it is derived from `inputValue` and we are already checking if inputValue starts with certain prefix

Reply:
Considering a generic component that doesn't know the logic of the checkSupported function, IMO we shouldn't show the unsupported warning if the field is empty.

Reply:
Ah gotcha, removing this is causing error on empty.